pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java

```diff
@@ -96,4 +96,10 @@ public String getDedupTimeColumn() {
   public boolean isEnablePreload() {
     return _enablePreload;
   }
+
+  // CHECKSTYLE:OFF
```

Comment:
Switched off the checkstyle as we would need to override server configs when extending

Reply:
This shouldn't be needed. Do you run into error without it?

Reply:
Yeah ran into checkstyle exception, like setter for enablePreload and enableSnapshot cannot be set in dedupConfig

Reply:
I know why it doesn't like it. This method should not return anything. Change the return to `void` should be good

Reply:
My bad, changed.